hk2-api/src/main/java/org/glassfish/hk2/utilities/RethrowErrorService.java

```diff
@@ -16,43 +16,41 @@
 
 package org.glassfish.hk2.utilities;
 
+import java.util.Objects;
+
 import javax.inject.Singleton;
 
 import org.glassfish.hk2.api.ErrorInformation;
 import org.glassfish.hk2.api.ErrorService;
 import org.glassfish.hk2.api.ErrorType;
 import org.glassfish.hk2.api.MultiException;
+import org.glassfish.hk2.utilities.reflection.Logger;
 
 /**
  * This is an implementation of {@link ErrorService} that simply rethrows
  * the exception caught.
  * <p>
- * By default HK2 ignores errors caught during a lookup operation.  This
- * service will make these errors get thrown up to the caller of the lookup
- * operation.
- * <p>
  * Do not use this service in secure applications where callers to lookup
  * should not be given the information that they do NOT have access
  * to a service.
- * 
- * @author jwells
  *
+ * @author jwells
+ * @author David Matejcek
  */
 @Singleton
 public class RethrowErrorService implements ErrorService {
 
-    /* (non-Javadoc)
-     * @see org.glassfish.hk2.api.ErrorService#onFailure(org.glassfish.hk2.api.ErrorInformation)
-     */
     @Override
-    public void onFailure(ErrorInformation errorInformation)
-            throws MultiException {
+    public void onFailure(ErrorInformation errorInformation) throws MultiException {
+        Logger.getLogger().debug("onFailure: " + errorInformation);
```

Comment:
The point of the rethrow error service is not to print out the error, but to rethrow it so it can be printed elsewhere, could you remove this line please.

Reply:
It is not redundant, because:

1. It sends log message to logger with error details, not stacktrace, it is useful for debugging.
2. Thrown exception is missing these information.

The only bad thing is that it invokes toString even if it is not needed. And HK2 logger does not support suppliers, so ok, I will remove it.